AsyncDisplayKit/ASCollectionView.mm

```diff
@@ -357,6 +382,14 @@ - (void)performBatchUpdates:(void (^)())updates completion:(void (^)(BOOL))compl
   [self performBatchAnimated:YES updates:updates completion:completion];
 }
 
+- (void)registerSupplementaryNodeOfKind:(NSString *)elementKind
+{
+  ASDisplayNodeAssert(elementKind != nil, @"A kind is needed for supplementary node registration");
+  [_registeredSupplementaryKinds addObject:elementKind];
+  [self registerClass:[UICollectionReusableView class] forSupplementaryViewOfKind:elementKind
```

Comment:
I could be wrong but I thought we had a custom subclass of UICollectionReusableView.  It may not actually do anything, although probably best to use it here.